operations/worker_pool.go

```diff
@@ -0,0 +1,59 @@
+package operations
+
+import (
+	"context"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/storage"
+	"time"
+)
+
+// WorkerPool is an abstraction responsible for processing
+// jobs which are scheduled by a JobScheduler
+type WorkerPool struct {
+	smCtx      context.Context
+	repository storage.Repository
+	jobs       chan ExecutableJob
+	workers    chan struct{}
+	jobTimeout time.Duration
+}
+
+// NewWorkerPool constructs a new worker pool
+func NewWorkerPool(ctx context.Context, repository storage.Repository, options *Settings) *WorkerPool {
+	return &WorkerPool{
+		smCtx:      ctx,
+		repository: repository,
+		jobs:       make(chan ExecutableJob, options.PoolSize),
+		workers:    make(chan struct{}, options.PoolSize),
+		jobTimeout: options.JobTimeout,
+	}
+}
+
+// Run starts the worker pool so it can start polling for scheduled jobs
+func (wp *WorkerPool) Run() {
+	go wp.processJobs()
```

Comment:
There are jobs also in the scheduler. No jobs are put in the worker pool. I suppose it is a typo?

Reply:
When constructing the scheduler we assign to it the worker pool's job channel. Both components communicate through it afterwards.